examples/stable-diffusion/unconditional_image_generation.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import logging
+
+from diffusers import DDPMScheduler
+from transformers.utils import check_min_version
+
+from optimum.habana.diffusers import GaudiDDIMScheduler, GaudiDDPMPipeline
+from optimum.habana.transformers.gaudi_configuration import GaudiConfig
+
+
+logger = logging.getLogger(__name__)
+
+try:
+    from optimum.habana.utils import check_optimum_habana_min_version
+except ImportError:
+
+    def check_optimum_habana_min_version(*a, **b):
+        return ()
+
+check_min_version("4.37.0")
+check_optimum_habana_min_version("1.10.4")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model_name_or_path",
+        default="google/ddpm-ema-celebahq-256",
+        type=str,
+        help="Path of the pre-trained unconditional image generation model",
+    )
+    parser.add_argument(
+        "--batch_size",
+        type=int,
+        default=16,
+        help="Batch size for the task.",
+    )
+    parser.add_argument(
+        "--num_inference_steps",
+        type=int,
+        default=1000,
+        help="Number of inference steps for the denoising UNet."
+    )
+    parser.add_argument(
+        "--use_gaudi_optimized_scheduler",
```

Comment:
Suggest renaming to `--use_gaudi_ddim_scheduler` so it's clear which scheduler is used.